browser/brave_vpn/brave_vpn_service_factory.cc

```diff
@@ -64,8 +73,15 @@ KeyedService* BraveVpnServiceFactory::BuildServiceInstanceFor(
       },
       context);
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
-  return new BraveVpnService(shared_url_loader_factory,
-                             user_prefs::UserPrefs::Get(context), callback);
+  auto* vpn_service = new BraveVpnService(
+      shared_url_loader_factory, user_prefs::UserPrefs::Get(context), callback);
+#if BUILDFLAG(IS_WIN)
+  auto* dns_observer_service =
+      brave_vpn::BraveVpnDnsObserverFactory::GetInstance()
+          ->GetServiceForContext(context);
+  dns_observer_service->Observe(vpn_service);
```

Comment:
Should this have a null check? (since `GetServiceForContext` has a path returning nullptr)

Reply:
done